src/cogs/help.py

```diff
@@ -0,0 +1,234 @@
+"""
+Mainly a help command that replaces the default command that discord.py provides.
+This module also contains related commands and functions such as about, contrib, docs.
+"""
+import configparser
+from contextlib import suppress
+
+import discord
+from discord.ext import commands
+
+
+def find_ci(dictionary, string):
+    """
+    Returns a value for a key in a dictionary. Unlike normal indexing
+    this is case insensitive, and returns None if no value exists
+    """
+    for key, value in dictionary.items():
+        if key.lower() == string.lower():
+            return value
+    return None
+
+
+def cog_values(self):
+    """Returns json for all cogs, commands and their related info."""
+    disabled_cmds = []
+    enabled_cmds = []
+    cog_dict = {}
+    misc_cmd_dict = {}
+    all_cmds = {}
+
+    for cog in self.bot.cogs:
+        current_cog_disabled_cmds = []
+        cmd_dict = {}
+        for cmd in self.bot.get_cog(cog).get_commands():
+            cmd_info = {
+                "name": cmd.name,
+                "usage": cmd.usage,
+                "doc_string": cmd.help,
+                "aliases": cmd.aliases,
+            }
+            cmd_item = {cmd.name: cmd_info}
+            cmd_dict[cmd.name] = cmd_info
+            all_cmds[cmd.name] = cmd_info
+
+            cmd_dict.update(cmd_item)
+            all_cmds.update(cmd_item)
+            if not cmd.enabled:
+                disabled_cmds.append(cmd.name)
+                current_cog_disabled_cmds.append(cmd.name)
+            else:
+                enabled_cmds.append(cmd.name)
+
+        cmd_count = len(self.bot.get_cog(cog).get_commands()) - len(
+            current_cog_disabled_cmds
+        )
```

Comment:
I'd probably throw self.bot.get_cog(cog).get_commands() in a variable at the top of the outer for loop, and use that for the inner for loops condition as well as this bit over here.

Reply:
That would increase the local variable limit over pylints maximum.